src/components/gallery-views/GalleryAllView.tsx

```diff
@@ -79,37 +81,34 @@ const GalleryAllView: React.FC<{ onLoadNextPage: () => Promise<void> }> = ({ onL
     }
   }
 
-  return (
-    <FlatList<PhotoWithPreview>
-      initialNumToRender={25}
-      getItemLayout={function (_, index) {
-        return {
-          index,
-          length: itemSize + GUTTER,
-          offset: (itemSize + GUTTER) * index,
-        };
-      }}
-      contentContainerStyle={{ paddingBottom: itemSize }}
-      style={tailwind('bg-white')}
-      showsVerticalScrollIndicator
-      indicatorStyle={'black'}
-      refreshControl={
-        <RefreshControl
-          refreshing={refreshing}
-          onRefresh={async () => {
-            setRefreshing(true);
+  async function handleRefresh() {
+    setRefreshing(true);
+    await onRefresh();
```

Comment:
What is going to happen if this method fails? Would the gallery keep loading infinitely instead of notifying the user?

Reply:
this method is handled by the parent, the parent calls a redux thunk

In case the method fails, the UI will keep the data in memory, so the old photos rendered will appear